agents/web-api/src/main/java/edu/jhuapl/dorset/agents/StockAgent.java

```diff
@@ -0,0 +1,277 @@
+    private HttpClient client;
+    private TreeMap<String, String> stockSymbolMap;
+    private static final int DAYS_IN_A_MONTH = 30;
+    private static final Logger logger = LoggerFactory.getLogger(StockAgent.class);
+   
+    private static final String SUMMARY =
+            "Stock ticker that gives historical stock information for NASDAQ and NYSE companies";
+
+    private static final CellProcessor[] processors = new CellProcessor[] {
+            new NotNull(), new NotNull(), new NotNull(), new Optional(),
+            new Optional(), new Optional(), new NotNull(), new Optional() };
+
+    /**
+     * Stock agent
+     *
+     * @param client An http client object
+     * @param apiKey A Quandl API key
+     */
+    public StockAgent(HttpClient client, String apiKey) {
+        this.client = client;
+        this.apiKey = apiKey;
+
+        ICsvBeanReader csvBeanReader = null;
+
+        InputStream nasdaqCompanies = StockAgent.class.getClassLoader().getResourceAsStream("stockagent/NASDAQ_Companies.csv");
+
+        this.stockSymbolMap = new TreeMap<String, String>(String.CASE_INSENSITIVE_ORDER);
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nasdaqCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+                this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NASDAQ_Companies.csv");
+        }
+
+        InputStream nyseCompanies = StockAgent.class.getClassLoader()
+                .getResourceAsStream("stockagent/NYSE_Companies.csv");
+
+        try {
+            csvBeanReader = new CsvBeanReader(new BufferedReader(new InputStreamReader(nyseCompanies)),
+                    CsvPreference.STANDARD_PREFERENCE);
+            final String[] header = csvBeanReader.getHeader(true);
+            CompanyInfo companyInfo;
+            while ((companyInfo = csvBeanReader.read(CompanyInfo.class, header, processors)) != null) {
+
+                if (!this.stockSymbolMap.containsKey(companyInfo.getName())) {
+                    this.stockSymbolMap.put(companyInfo.getName(), companyInfo.getSymbol());
+                }
+            }
+
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            logger.debug("Failed to load stockagent/NYSE_Companies.csv");
+        } 
+
+    }
+
+    @Override
+    public AgentResponse process(AgentRequest request) {
+        logger.debug("Handling the request: " + request.getText());
+
+        String requestCompanyName = request.getText();
+        String[] stockCompanyInfo = findStockSymbol(requestCompanyName);
```

Comment:
I recommend returning the map entry, creating a small class to hold the name and symbol, or returning the entire CompanyInfo object (which would mean storing that in the map).

Any of those options are more explicit rather than having to know that the name is index 0 and the symbol index 1